### PR TITLE
Add the `bindgen` Users Guide

### PR DESCRIPTION
**Pull request by fitzgen**
This commit adds a Users Guide, built with [mdbook][] (the same tool used
by *The Rust Book*).

It moves all the user documentation from the README.md into the Users Guide. It
leaves the README.md as a splash page featuring an example of what `bindgen` is
and does, a link to the Users Guide, a link to the docs.rs/bindgen API reference
documentation, and a link to CONTRIBUTING.md.

It ensures that the Users Guide builds with `mdbook`, and all code snippets
within it build as doctests and pass, on every CI run.

Finally, it builds and uploads the Users Guide every time CI passes on the
master branch.

[mdbook]: https://github.com/azerupi/mdBook

r? @emilio see the rendered book here: https://servo.github.io/rust-bindgen/index.html

### CODE REVIEW

File: book/src/requirements.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+greater.
+
+### Installing Clang 3.9
+
+#### Windows
+
+Download and install the official pre-built binary from
+[LLVM download page](http://releases.llvm.org/download.html).
+
+#### OSX
```
Comment (atouchet) [inline book/src/requirements.md]: Apple no longer uses the OS X branding. Can this be changed to say macOS?

Comment (fitzgen) [inline book/src/requirements.md]: I just copied this from the README, but yeath that's a good point.

File: book/src/requirements.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+greater.
+
+### Installing Clang 3.9
+
+#### Windows
+
+Download and install the official pre-built binary from
+[LLVM download page](http://releases.llvm.org/download.html).
+
+#### MacOS
```
Comment (atouchet) [inline book/src/requirements.md]: I don't know how much it really matters but Apple's branding uses a lowercase m if you want to follow their guidelines.

Comment (fitzgen) [inline book/src/requirements.md]: Fixed

**Comment by emilio** [comment]
Looks quite awesome, thanks _a lot_ for doing this!

@bors-servo r+

**Pull request merged**